guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/upload/UploadFormPresenter.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.m2repo.client.upload;
+
+import static org.guvnor.m2repo.model.HTMLFileManagerFields.UPLOAD_MISSING_POM;
+import static org.guvnor.m2repo.model.HTMLFileManagerFields.UPLOAD_OK;
+import static org.guvnor.m2repo.model.HTMLFileManagerFields.UPLOAD_UNABLE_TO_PARSE_POM;
+import static org.guvnor.m2repo.utils.FileNameUtilities.isValid;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+
+import com.github.gwtbootstrap.client.ui.Form.SubmitCompleteEvent;
+import com.github.gwtbootstrap.client.ui.Form.SubmitEvent;
+import com.github.gwtbootstrap.client.ui.event.HideEvent;
+import com.github.gwtbootstrap.client.ui.event.HideHandler;
+import org.guvnor.m2repo.client.event.M2RepoSearchEvent;
+import org.jboss.errai.ioc.client.container.SyncBeanManager;
+
+@Dependent
+public class UploadFormPresenter implements UploadFormView.Presenter {
+
+    Event<M2RepoSearchEvent> searchEvent;
+
+    private UploadFormView view;
+
+    @Inject
```

Comment:
Can you please move this to the constructor too (#1 for consistency, #2 you'll be able to write a test for "closing" the upload form and check ```SyncBeanManager.destroyBean()``` is called)?